qiskit/extensions/hamiltonian_gate.py

```diff
@@ -27,9 +27,12 @@
 
 
 class HamiltonianGate(Gate):
-    """Class for representing evolution by a Hermitian Hamiltonian operator as a gate. This gate
-    resolves to a UnitaryGate U(t) = exp(-1j * t * H), which can be decomposed into basis gates if
-    it is 2 qubits or less, or simulated directly in Aer for more qubits."""
+    """Class for representing evolution by a Hermitian Hamiltonian operator as
+    a gate. This gate resolves to a :class:`UnitaryGate` as :math:`U(t) =
+    exp(-1j * t * H)`, which can be decomposed into basis gates if it is 2
+    qubits or less, or simulated directly in Aer for more qubits. Note that you
+    can also directly use :meth:`~qiskit.circuit.QuantumCircuit.hamiltonian`.
+    """
```

Comment:
With a single header line this renders nicely in the docs 🙂 Also in math mode we don't need to use Python notation 😄 
```suggestion
    """Class for representing evolution by a Hamiltonian operator as a gate.
    
    This gate resolves to a :class:`.UnitaryGate` as :math:`U(t) =
    exp(-i t H)`, which can be decomposed into basis gates if it is 2
    qubits or less, or simulated directly in Aer for more qubits. Note that you
    can also directly use :meth:`.QuantumCircuit.hamiltonian`.
    """
```